### PR TITLE
CDAP 19700 Workspace List Page

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE
Feature - Workspace List Page

## Description
Summary of changes
This PR contains the following changes
1. Workspace List Page
2. Ongoing Explorations (code taken from another PR for completeness: https://github.com/cdapio/cdap-ui/pull/729)

## Please find the List of Files to be reviewed below
1. app/cdap/components/WrangleHome/index.tsx
2. app/cdap/components/WorkspaceList/*
3. app/cdap/components/Home/index.js
4. app/cdap/text/text-en.yaml
5. src/e2e-test/*

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19700

## Test Plan
1. FE_04_025 - Verify the "Ongoing data exploration" Section on the Home Page is hidden or not when we don't have any connections loaded in the grid page. - Pass
2. FE_04_026 - Verify the "view all" text is displayed beside the 'Continue ongoing exploration, pick up where you left on the Home Page or not - Pass
3. FE_04_027 - Verify the "view all" text by mouse hovering. - Pass
4. FE_04_028 - Verify the "view all" text by Clicking. - Pass
5. FE_04_029 - Verify the redirection after clicking on the "view all" text link. - Pass
6. FE_04_030 - Verify whether the workspaces are listed as per design or not. - Pass
7. FE_04_032 - Verify the listed workspace by mouse hover - Pass
8. FE_04_033 - Verify the selected workspace list is redirected to a particular selected list. - Pass

## Screenshots
Screenshot 2022-11-09 at 8 35 13 PM

Screenshot 2022-11-09 at 8 34 56 PM

### CODE REVIEW

**Review by seanfeld — COMMENTED** [review]
Are there duplicate code changes in here? if there are duplicate changes I need to wait to review this since its 2800+ lines

File: app/cdap/components/WorkspaceList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    if (
+      !(
+        DataPrepStore.getState().dataprep.connectorsWithIcons &&
+        DataPrepStore.getState().dataprep.connectorsWithIcons.length > 0
+      )
+    ) {
+      getWidgetData();
+    }
+  }, []);
+  const classes = useStyles();
```
Comment (njbriggs) [inline app/cdap/components/WorkspaceList/index.tsx]: Please use `styled-components`

File: app/cdap/text/text-en.yaml [hunk 2]
```diff
@@ -102,25 +102,25 @@ features:
     copyright:
       firstLine: Copyright © 2014-2017 Cask Data, Inc.
       secondLine:
-        view: "View "
+        view: 'View '
```
Comment (njbriggs) [inline app/cdap/text/text-en.yaml]: Please don't make unrelated changes in this file

File: app/cdap/components/WorkspaceList/index.tsx (hunk 1 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/WorkspaceList/index.tsx]: Done. 
Removed all the elements with classnames & replaced them with styled components.

File: app/cdap/text/text-en.yaml (hunk 2 above)
Comment (sivaRajanaDivami) [reply app/cdap/text/text-en.yaml]: Done

**Comment by sivaRajanaDivami** [comment]
> Are there duplicate code changes in here? if there are duplicate changes I need to wait to review this since its 2800+ lines

We have added the files which need to be reviewed in the description of this PR.
Please check and let us know if you need more information.
